File: mobile/src/components/ListCard.js
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { colors, radii, spacing, shadows } from '../theme/colors';

/**
 * بطاقة موحدة بنمط التصميم الجديد
 *  ──────────────────────────────────────
 *  | title            topRight (small)  |
 *  |                                    |
 *  | subtitleA                          |
 *  | subtitleB                 amount   |
 *  ──────────────────────────────────────
 */
export default function ListCard({
  title,
  topRight,
  subtitleA,
  subtitleB,
  amount,
  amountColor,
  onPress,
  style,
}) {
  const Wrapper = onPress ? TouchableOpacity : View;
  return (
    <Wrapper activeOpacity={0.85} onPress={onPress} style={[styles.card, style]}>
      <View style={styles.row}>
        <Text style={styles.title} numberOfLines={1}>{title}</Text>
        {topRight ? <Text style={styles.topRight} numberOfLines={1}>{topRight}</Text> : null}
      </View>

      <View style={styles.bottomRow}>
        <View style={{ flex: 1 }}>
          {subtitleA ? <Text style={styles.subtitle}>{subtitleA}</Text> : null}
          {subtitleB ? <Text style={styles.subtitle}>{subtitleB}</Text> : null}
        </View>
        {amount != null && (
          <Text style={[styles.amount, amountColor && { color: amountColor }]}>{amount}</Text>
        )}
      </View>
    </Wrapper>
  );
}

const styles = StyleSheet.create({
  card: {
    backgroundColor: colors.card,
    borderRadius: radii.lg,
    padding: spacing.lg,
    ...shadows.soft,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    gap: spacing.md,
  },
  title: {
    fontSize: 15,
    fontWeight: '700',
    color: colors.text,
    flex: 1,
    textAlign: 'right',
    letterSpacing: -0.2,
  },
  topRight: {
    fontSize: 12,
    color: colors.textSecondary,
    fontWeight: '500',
  },
  bottomRow: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'space-between',
    marginTop: spacing.md,
    gap: spacing.md,
  },
  subtitle: {
    fontSize: 12,
    color: colors.textMuted,
    textAlign: 'right',
    lineHeight: 18,
  },
  amount: {
    fontSize: 15,
    fontWeight: '700',
    color: colors.text,
    letterSpacing: -0.2,
  },
});

File: mobile_flutter/lib/widgets/list_card.dart
import 'package:flutter/material.dart';
import '../theme/colors.dart';

class ListCard extends StatelessWidget {
  final String title;
  final String? topRight;
  final String? subtitleA;
  final String? subtitleB;
  final String? amount;
  final Color? amountColor;
  final VoidCallback? onTap;

  const ListCard({
    super.key,
    required this.title,
    this.topRight,
    this.subtitleA,
    this.subtitleB,
    this.amount,
    this.amountColor,
    this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final body = Container(
      padding: const EdgeInsets.all(AppSpacing.lg),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: BorderRadius.circular(AppRadii.lg),
        boxShadow: AppShadows.soft,
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Expanded(
                child: Text(
                  title,
                  style: const TextStyle(
                    fontSize: 15,
                    fontWeight: FontWeight.w700,
                    color: AppColors.text,
                    letterSpacing: -0.2,
                  ),
                  textAlign: TextAlign.right,
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                ),
              ),
              if (topRight != null) ...[
                const SizedBox(width: AppSpacing.md),
                Text(
                  topRight!,
                  style: const TextStyle(
                    fontSize: 12,
                    color: AppColors.textSecondary,
                    fontWeight: FontWeight.w500,
                  ),
                ),
              ],
            ],
          ),
          const SizedBox(height: AppSpacing.md),
          Row(
            crossAxisAlignment: CrossAxisAlignment.end,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  children: [
                    if (subtitleA != null)
                      Text(subtitleA!,
                          textAlign: TextAlign.right,
                          style: const TextStyle(fontSize: 12, color: AppColors.textMuted, height: 1.5)),
                    if (subtitleB != null)
                      Text(subtitleB!,
                          textAlign: TextAlign.right,
                          style: const TextStyle(fontSize: 12, color: AppColors.textMuted, height: 1.5)),
                  ],
                ),
              ),
              if (amount != null) ...[
                const SizedBox(width: AppSpacing.md),
                Text(
                  amount!,
                  style: TextStyle(
                    fontSize: 15,
                    fontWeight: FontWeight.w700,
                    color: amountColor ?? AppColors.text,
                    letterSpacing: -0.2,
                  ),
                ),
              ],
            ],
          ),
        ],
      ),
    );

    if (onTap == null) return body;
    return Material(
      color: Colors.transparent,
      child: InkWell(
        onTap: onTap,
        borderRadius: BorderRadius.circular(AppRadii.lg),
        child: body,
      ),
    );
  }
}
